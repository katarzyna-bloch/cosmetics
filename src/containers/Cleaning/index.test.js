import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Cleaning } from '.'

jest.mock('../../components/ProductsList', () => {
  const ProductsList = () => null
  return ProductsList
})

it('should render correctly', () => {
  const props = {
    fetchProducts: jest.fn(),
    products: [
      { name: 'Test', img: 'img.jpg', id: '1', price: '10z≈Ç' },
    ],
  }
  const wrapped = mount(<Cleaning {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()
  wrapped.update()
  expect(props.fetchProducts).toHaveBeenCalled()
})
