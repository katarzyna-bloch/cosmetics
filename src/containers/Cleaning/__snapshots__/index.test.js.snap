// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Cleaning
  fetchProducts={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  products={
    Array [
      Object {
        "id": "1",
        "img": "img.jpg",
        "name": "Test",
        "price": "10zł",
      },
    ]
  }
>
  <styled.div>
    <div
      className="sc-bdfBwQ lbYnYk"
    >
      <styled.h5>
        <h5
          className="sc-gsTCUz jUlaeV"
        >
          Oczyszczanie to podstawowy krok w dojściu do pięknej skóry
        </h5>
      </styled.h5>
      <ProductsList
        products={
          Array [
            Object {
              "id": "1",
              "img": "img.jpg",
              "name": "Test",
              "price": "10zł",
            },
          ]
        }
      >
        <styled.div>
          <div
            className="sc-iBPRYJ jdskZy"
          >
            <Product
              img="img.jpg"
              key="1"
              name="Test"
              price="10zł"
            >
              <styled.div>
                <div
                  className="sc-gKsewC ksRYnp"
                >
                  <styled.div>
                    <div
                      className="sc-hKgILt cLowGV"
                    >
                      <styled.img
                        alt="cosmetic"
                        src="img.jpg"
                      >
                        <img
                          alt="cosmetic"
                          className="sc-dlfnbm blFtGQ"
                          src="img.jpg"
                        />
                      </styled.img>
                      <styled.div>
                        <div
                          className="sc-eCssSg kWxpnN"
                        >
                          Test
                        </div>
                      </styled.div>
                      <styled.div>
                        <div
                          className="sc-jSgupP bjHwBQ"
                        >
                          10zł
                        </div>
                      </styled.div>
                    </div>
                  </styled.div>
                </div>
              </styled.div>
            </Product>
          </div>
        </styled.div>
      </ProductsList>
    </div>
  </styled.div>
</Cleaning>
`;
